lib/customer-deposit-wallet/src/Cardano/Wallet/Deposit/REST.hs

```diff
@@ -369,6 +370,10 @@ createAddress
     -> WalletResourceM Address
 createAddress = onWalletInstance . WalletIO.createAddress
 
+customerAddress :: Customer -> WalletResourceM (Maybe Address)
+customerAddress c =
+    lookup c <$> listCustomers
```

Comment:
```suggestion
customerAddress = onWalletInstance . WalletIO.customerAddress
```

Could you move this all the way down to `Pure`?

Having a function

```hs
customerAddress :: Customer → Maybe Address
```

with the (temporary, because not performant) implementation

```hs
customerAddress c = lookup c <$> listCustomers
```

as part of the pure Haskell interface is perfectly valid.